### DISCUSSION TITLE
Issues with relay pagination and order

### DISCUSSION THREAD

**Discussion by dih7** [hayes/pothos#1188] [Q&A]
Hello,

I'm having an issue where the pagination doesn't work when the orderBy is set. I'm using the Prisma & Relay plugins.

This is the Page Prisma node object;

```
export const Page = builder.prismaNode("Page", {
  id: { field: "id" },
  fields: (t) => ({
    createdAt: t.expose("createdAt", { type: "DateTime" }),
    updatedAt: t.expose("updatedAt", { type: "DateTime" }),
    title: t.exposeString("title"),
    slug: t.exposeString("slug"),
    excerpt: t.exposeString("excerpt", { nullable: true }),
    published: t.exposeBoolean("published"),
    order: t.exposeFloat("order", { nullable: true }),
    // ... other fields
  }),
});
```

Here is a section of the Pothos query field object;

```
  userPages: t.prismaConnection({
    type: "Page",
    authScopes: {
      activeSession: true,
    },
    cursor: "id",
    args: {
      pageType: t.arg.string(),
    },
    resolve: (query, _, { pageType }) =>
      db.page.findMany({
        ...query,
        // orderBy: {
        //   order: "asc",
        // },
        where: {
          published: true,
          ...(pageType
            ? {
                type: pageType,
              }
            : {}),
        },
        include: {
          content: {
            orderBy: {
              order: "asc",
            },
          },
        },
      }),
  }),
```

The pagination works as intended when the orderBy section is commented out, but behaves strangely when the orderBy is enabled. The strange behaviour occurs when going to the last page of results and navigating back - certain items start disappearing, preventing the pagination from working.

Is there a glaring issue I'm missing here?

I appreciate that this may not be much to go off, but thanks in advance!

**Comment by hayes** [comment]
~The query object being spread in includes an order by that you are overwriting~ actually,.that's not the issue... Misremembered how this works

**Reply by hayes** [reply]
I can't see anything obvious here. I'd recommend logging out the fully query being passed to Prisma to see if there is anything in there that seems unexpected.  I'd probably need more details on what's going on with the missing results (what data is returned/missing, what the actual queries are that are being executed)
